src/EnergyPlus/HeatBalanceManager.cc

```diff
@@ -4510,73 +4510,71 @@ namespace HeatBalanceManager {
-                state.dataMaterial->Material(loop)->ReflectSolBeamBack = 0.0;
-                state.dataMaterial->Material(loop)->ReflectSolBeamFront = 0.0;
-                state.dataMaterial->Material(loop)->ReflectSolDiffBack = 0.0;
-                state.dataMaterial->Material(loop)->ReflectSolDiffFront = 0.0;
-                state.dataMaterial->Material(loop)->ReflectVisBeamBack = 0.0;
-                state.dataMaterial->Material(loop)->ReflectVisBeamFront = 0.0;
-                state.dataMaterial->Material(loop)->ReflectVisDiffBack = 0.0;
-                state.dataMaterial->Material(loop)->ReflectVisDiffFront = 0.0;
-                state.dataMaterial->Material(loop)->TransSolBeam = 0.0;
-                state.dataMaterial->Material(loop)->TransThermal = 0.0;
-                state.dataMaterial->Material(loop)->TransVisBeam = 0.0;
-                state.dataMaterial->Material(loop)->GlassSpectralDataPtr = 0;
-                state.dataMaterial->Material(loop)->NumberOfGasesInMixture = 0;
-                state.dataMaterial->Material(loop)->GasCon = 0.0;
-                state.dataMaterial->Material(loop)->GasVis = 0.0;
-                state.dataMaterial->Material(loop)->GasCp = 0.0;
-                state.dataMaterial->Material(loop)->gasTypes = Material::GasType::Custom;
-                state.dataMaterial->Material(loop)->GasWght = 0.0;
-                state.dataMaterial->Material(loop)->GasSpecHeatRatio = 0.0;
-                state.dataMaterial->Material(loop)->GasFract = 0.0;
-                state.dataMaterial->Material(loop)->WinShadeToGlassDist = 0.0;
-                state.dataMaterial->Material(loop)->WinShadeTopOpeningMult = 0.0;
-                state.dataMaterial->Material(loop)->WinShadeBottomOpeningMult = 0.0;
-                state.dataMaterial->Material(loop)->WinShadeLeftOpeningMult = 0.0;
-                state.dataMaterial->Material(loop)->WinShadeRightOpeningMult = 0.0;
-                state.dataMaterial->Material(loop)->WinShadeAirFlowPermeability = 0.0;
-                state.dataMaterial->Material(loop)->BlindDataPtr = 0;
-                state.dataMaterial->Material(loop)->EMPDmu = 0.0;
-                state.dataMaterial->Material(loop)->MoistACoeff = 0.0;
-                state.dataMaterial->Material(loop)->MoistBCoeff = 0.0;
-                state.dataMaterial->Material(loop)->MoistCCoeff = 0.0;
-                state.dataMaterial->Material(loop)->MoistDCoeff = 0.0;
-                state.dataMaterial->Material(loop)->EMPDSurfaceDepth = 0.0;
-                state.dataMaterial->Material(loop)->EMPDDeepDepth = 0.0;
-                state.dataMaterial->Material(loop)->EMPDmuCoating = 0.0;
-                state.dataMaterial->Material(loop)->EMPDCoatingThickness = 0.0;
+                auto *thisMaterial = new Material::MaterialChild;
+                state.dataMaterial->Material.push_back(thisMaterial);
+                thisMaterial->Name = "";
```

Comment:
If these are the default values for everything (i.e., they are what the default constructor would put in) then you don't need this code. The default constructor is called by `new`.

Reply:
very true. The values are the same as the default values.